feat(frontend): add AI Question Paper Generator card and form

dkhushi6 opened the pull request:
- Added a new card on the homepage linking to the Question Paper Generator
- Built the frontend route for the generator
- Implemented form with sections, topics, and question types
- Added header and description text for better context

## 📋 Pull Request

### 🔗 Related Issue
Fixes #

### 📝 What changes did you make?
<!-- Briefly describe your changes -->


### 📸 Screenshots/Screen Recording 
<!-- Drag and drop images/videos here or paste links -->


### ✅ Checklist
- [ ] ⭐ I have starred the repository (Mandatory for GSSoC 25)

- [ ] 🧪 I have tested my changes
- [ ] 📖 I have read the [Contribution Guidelines](https://github.com/huamanraj/aitoolbox.softwar
<img width="1707" height="980" alt="Screenshot 2025-09-24 at 5 09 45 PM" src="https://github.com/user-attachments/assets/f35aba7c-e874-4238-8380-e2fd5145b4c9" />
e/tree/main/contributions)

---
<img width="1706" height="980" alt="Screenshot 2025-09-24 at 5 09 21 PM" src="https://github.com/user-attachments/assets/36b47046-9da6-4642-8422-6c8fcd5b16ce" />
<img width="1710" height="983" alt="Screenshot 2025-09-24 at 5 09 33 PM" src="https://github.com/user-attachments/assets/3e325fde-aaed-482e-a90d-4af70cea16d5" />

<img width="1710" height="983" alt="Screenshot 2025-09-24 at 5 10 27 PM" src="https://github.com/user-attachments/assets/86b1ecb7-aee8-4ff8-826e-128105529bbd" />

*Thank you for contributing to AIToolbox.software!🌸*

huamanraj commented:
good work @dkhushi6 👏 

just some more improvements;
- Make it mobile responsive 
- add it to the sidebar
- Make sure it is SSR

dkhushi6 commented:
@huamanraj 
Completed the backend implementation and incorporated the changes you suggested. Kindly review the PR and provide your remarks.

* Implemented `/api/question-paper` route to generate AI-powered question papers using Gemini API.
* Added server-side handling with rate limiting and robust error management.
* Connected frontend `QuestionPaperForm` to fetch API via SSR.
* Implemented loading, progress, and error states for better user experience.
* Integrated AI-generated output display in a collapsible section.
* Added the Question Paper route to the sidebar for quick navigation.
* Updated layout and styling for mobile responsiveness.

**Setup Instructions:**

* Add your Gemini API key in `.env.local` as `GEMINI_API_KEY` to enable AI generation.
* Run the app and use the form to generate question papers.

Please review the changes and provide your feedback.
